envi/archs/arm/disasm.py

```diff
@@ -51,89 +34,99 @@ def addrToName(mcanv, va):
         return repr(sym)
     return "0x%.8x" % va
 
+
 # The keys in this table are made of the
 # concat of bits 27-21 and 7-4 (only when
 # ienc == mul!
 iencmul_codes = {
     # Basic multiplication opcodes
-    binary("000000001001"): ("mul",(0,4,2), 0),
-    binary("000000011001"): ("mul",(0,4,2), IF_PSR_S),
-    binary("000000101001"): ("mla",(0,4,2,1), 0),
-    binary("000000111001"): ("mla",(0,4,2,1), IF_PSR_S),
-    binary("000001001001"): ("umaal",(1,0,4,2), 0),
-    binary("000010001001"): ("umull",(1,0,4,2), 0),
-    binary("000010011001"): ("umull",(1,0,4,2), IF_PSR_S),
-    binary("000010101001"): ("umlal",(1,0,4,2), 0),
-    binary("000010111001"): ("umlal",(1,0,4,2), IF_PSR_S),
-    binary("000011001001"): ("smull",(1,0,4,2), 0),
-    binary("000011011001"): ("smull",(1,0,4,2), IF_PSR_S),
-    binary("000011101001"): ("smlal",(1,0,4,2), 0),
-    binary("000011111001"): ("smlal",(1,0,4,2), IF_PSR_S),
+    0b000000001001: ("mul",   INS_MUL,  (0, 4, 2), 0),
```

Comment:
to my understanding, the INS_* has always been intended to be an improved method of instruction emulation (determining what an emulator function is based on numbers rather than strings). at least, i know visi had mentioned the desire to not have emulation based on mnemonic, and i believe he meant to have it based on INS_* constat.
that sounds like a good conversation to have.